### PR TITLE
Added the functionality to reuse cache in app_runner

### PR DESCRIPTION
**Pull request by mkamyab**
### Rationale
For integrated testing of scheduled or stream apps, we are relying on the cache to be reused. However, the `app_runner` does not provide handlers to support this functionality. I'm suggesting adding a `kwarg` in the `app_runner` to support this.

### Changes
- Added a `kwarg` to `app_runner` plugin
- Added a test case for this feature

#### TODO
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: src/corva/testing.py [hunk 1]
```diff
@@ -36,7 +36,8 @@ def run(
         fn: Callable,
         event: Union[TaskEvent, StreamEvent, ScheduledEvent],
         *,
-        secrets: Optional[Dict[str, str]] = None
+        secrets: Optional[Dict[str, str]] = None,
+        cache: UserRedisSdk = None
```
Comment (oleksii-symon-corva-ai) [inline src/corva/testing.py:40]:
```suggestion
        cache: Optional[UserRedisSdk] = None
```

**Review by oleksii-symon-corva-ai — APPROVED** [review]
looks good!

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [1.5.4] - 2022-10-19
+
+### Changed
+- Added the ability to `app_runner` to reuse the cache object.
```
Comment (oleksii-symon-corva-ai) [inline CHANGELOG.md:13]: This is still unreleased

```suggestion
## [Unreleased]

### Changed
- Added `cache` parameter to `app_runner` to reuse the cache object. 
```

File: setup.cfg [hunk 3]
```diff
@@ -25,6 +25,7 @@ omit =
     docs/modules/ROOT/examples/logging/tutorial003.py
     docs/modules/ROOT/examples/logging/tutorial004.py
     docs/modules/ROOT/examples/logging/tutorial005.py
+    docs/modules/ROOT/examples/testing/tutorial008.py
```
Comment (oleksii-symon-corva-ai) [inline setup.cfg:28]:
```suggestion
```

Examples in docs showcase use cases of the system. We run tests against those examples to make sure they work as expected. `logging/tutorial003 - 005` examples were skipped from tests becase they use external systems like `rollbar` and `sentry`, which can not really be tested.

The point is we should run tests from `docs/modules/ROOT/examples/testing/tutorial008.py` in `tests.unit.test_docs.test_testing`. The test would look like following:
```python
def test_tutorial008(app_runner):
    tutorial008.test_reset_cache(app_runner)
    tutorial008.test_reuse_cache(app_runner)
```

Also after writing test above there is no need to have `test_reuse_cache` in `test_plugin`, which is a duplicate, so it can be removed.

**Review by oleksii-symon-corva-ai — COMMENTED** [review]
Great job, last stretch left and good to go!

File: CHANGELOG.md (hunk 2 above)
Comment (mkamyab) [reply CHANGELOG.md:13]: Thanks ... done

File: setup.cfg (hunk 3 above)
Comment (mkamyab) [reply setup.cfg:28]: Done

**Comment by oleksii-symon-corva-ai** [comment]
Looks good, let's merge!
